### PR TITLE
Scoreboard: Rework and MIA/Dead re-positioning fix

### PR DESCRIPTION
**Pull request by Alf21**

### CODE REVIEW

File: code/ui/generalhud/scoreboard/Scoreboard.ScoreboardGroup.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                            group = DefaultScoreboardGroup.Dead.ToString();
+                        }
+                        else if (player.IsMissingInAction)
+                        {
+                            group = DefaultScoreboardGroup.MIA.ToString();
+                        }
+
+                        break;
+                    }
+                }
```
Comment (mzegar) [inline code/ui/generalhud/scoreboard/Scoreboard.ScoreboardGroup.cs:101]: Might be able to use our "GetAllPlayers" util method here.

**Review by mzegar — COMMENTED** [review]
Minor comment

File: code/ui/generalhud/scoreboard/Scoreboard.ScoreboardGroup.cs (hunk 1 above)
Comment (Alf21) [reply code/ui/generalhud/scoreboard/Scoreboard.ScoreboardGroup.cs:101]: Uhm do you have a link to that method? I don't find anyone that would make sense to be integrated here sadly :/

Comment (mzegar) [reply code/ui/generalhud/scoreboard/Scoreboard.ScoreboardGroup.cs:101]: nvm, doesn't make sense in this case.

**Review by mzegar — APPROVED** [review]
Seems reasonable to me? I like that we threw the header into a new file. It was getting a bit long.

**Pull request merged**